src/main/java/com/farion/onlinebookstore/util/ConditionHolder.java

```diff
@@ -0,0 +1,14 @@
+package com.farion.onlinebookstore.util;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.stereotype.Component;
+
+@Setter
+@Getter
+@Component
+public class ConditionHolder {
+    private Set<Long> condition = new HashSet<>();
```

Comment:
Why this is required? To make it work properly it is required to use ThreadLocal for holder